catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroSettingsActivityTest.java

```diff
@@ -0,0 +1,172 @@
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_MINDSTORMS_NXT_BRICKS_ENABLED;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_MINDSTORMS_NXT_SHOW_SENSOR_INFO_BOX_DISABLED;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_ARDUINO_BRICKS;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_HINTS;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_NFC_BRICKS;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_PARROT_AR_DRONE_BRICKS;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_PHIRO_BRICKS;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_RASPI_BRICKS;
+import static org.hamcrest.Matchers.allOf;
+
+@RunWith(AndroidJUnit4.class)
+public class PhiroSettingsActivityTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<PhiroSettingsActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(PhiroSettingsActivity.class, true, false);
+
+	private List<String> allSettings = new ArrayList<>(Arrays.asList(
+			SETTINGS_SHOW_ARDUINO_BRICKS, SETTINGS_SHOW_NFC_BRICKS, SETTINGS_SHOW_HINTS, SETTINGS_CRASH_REPORTS,
+			SETTINGS_MINDSTORMS_NXT_BRICKS_ENABLED, SETTINGS_MINDSTORMS_NXT_SHOW_SENSOR_INFO_BOX_DISABLED,
+			SETTINGS_MINDSTORMS_EV3_BRICKS_ENABLED, SETTINGS_MINDSTORMS_EV3_SHOW_SENSOR_INFO_BOX_DISABLED,
+			SETTINGS_SHOW_PARROT_AR_DRONE_BRICKS, SETTINGS_SHOW_RASPI_BRICKS, SETTINGS_CAST_GLOBALLY_ENABLED));
+
+	private Map<String, Boolean> initialSettings = new HashMap<>();
+
+	@Before
+	public void setUp() throws Exception {
+		SharedPreferences sharedPreferences = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+
+		for (String setting : allSettings) {
+			initialSettings.put(setting, sharedPreferences.getBoolean(setting, false));
+		}
+		initialSettings.put(SETTINGS_SHOW_PHIRO_BRICKS, sharedPreferences.getBoolean(SETTINGS_SHOW_PHIRO_BRICKS, false));
+
+		setAllSettingsTo(true);
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	private void setAllSettingsTo(boolean value) {
+		SharedPreferences.Editor sharedPreferencesEditor = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext()).edit();
+
+		for (String setting : allSettings) {
+			sharedPreferencesEditor.putBoolean(setting, value);
+		}
+		sharedPreferencesEditor.apply();
+	}
+
+	@After
+	public void tearDown() {
+		SharedPreferences.Editor sharedPreferencesEditor = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext()).edit();
+		for (String setting : initialSettings.keySet()) {
+			sharedPreferencesEditor.putBoolean(setting, initialSettings.get(setting));
+		}
+		sharedPreferencesEditor.apply();
+		initialSettings.clear();
+	}
+
+	@Test
+	public void basicSettingsTest() {
+		checkPreference(R.string.preference_title_enable_arduino_bricks, SETTINGS_SHOW_ARDUINO_BRICKS);
```

Comment:
this is already tested in SettingsActivityTest, is there any point in duplicating this test?